### ISSUE TITLE
Failed to get component file

### ISSUE THREAD

**Issue by dungdm93** [gabrie-allaigre/sonar-gitlab-plugin#146]
I got following error while run `sonar-scanner` inside **GitLab-CI**:
```log
INFO: ANALYSIS SUCCESSFUL, you can browse http://sonarqube.teko.vn/dashboard/index/vn.teko.marketing:notification-client
INFO: Note that you will be able to access the updated dashboard once the server has processed the submitted analysis report
INFO: More about the report processing at http://sonarqube.teko.vn/api/ce/task?id=AWO1fh5lTiKdFtP2KoRf
INFO: Executing post-job GitLab Commit Issue Publisher
INFO: Waiting quality gate to complete...
INFO: Waiting quality gate to complete...
INFO: Waiting quality gate to complete...
INFO: Waiting quality gate to complete...
INFO: Waiting quality gate to complete...
INFO: Quality gate status: ERROR
INFO: Security Rating on New Code : 1
INFO: Reliability Rating on New Code : 1
INFO: Maintainability Rating on New Code : 1
ERROR: Coverage on New Code: 0.0 < 80
INFO: Duplicated Lines on New Code (%) : 0.0
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 15.016s
INFO: Final Memory: 16M/397M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarQube Scanner execution
ERROR: SonarQube failed to complete the review of this commit: Failed to get component file for vn.teko.marketing:notification-client:src/reducers/customerCount.js
ERROR: Caused by: Failed to get component file for vn.teko.marketing:notification-client:src/reducers/customerCount.js
ERROR: Caused by: org.sonarqube.ws.client.HttpException: Error 404 on http://sonarqube.teko.vn/api/components/show?key=vn.teko.marketing:notification-client:src/reducers/customerCount.js
ERROR: Caused by: Error 404 on http://sonarqube.teko.vn/api/components/show?key=vn.teko.marketing:notification-client:src/reducers/customerCount.js
ERROR: 
ERROR: Re-run SonarQube Scanner using the -X switch to enable full debug logging.
ERROR: Job failed: exit code 1
```
I realize that erorr is occur on **non-master branch** which contains **new files**.
For example, my case above, I run on `develop` branch and `src/reducers/customerCount.js` is new file.

```yaml
# .gitlab-ci.yaml
quality:sonarqube:
  stage: test
  image: dungdm93/sonar-scanner:3.2-alpine
  cache:
    key: sonar-cache
    paths: [ .sonar ]
  variables:
    SONAR_USER_HOME: ${CI_PROJECT_DIR}/.sonar
  script:
    - |
      case $CI_COMMIT_REF_NAME in
        "master" | "develop" | release/*) TARGET_BRANCH="master";;
        *) TARGET_BRANCH="develop";;
      esac
    - sonar-scanner
        -Dsonar.host.url=http://sonarqube.teko.vn
        -Dsonar.login=$SONAR_TOKEN
        -Dsonar.branch.name=$CI_COMMIT_REF_NAME
        -Dsonar.branch.target=$TARGET_BRANCH
        -Dsonar.gitlab.project_id=$CI_PROJECT_ID
        -Dsonar.gitlab.commit_sha=$CI_COMMIT_SHA
        -Dsonar.gitlab.ref_name=$CI_COMMIT_REF_NAME
```

**Comment by gabrie-allaigre** [comment]
Hi,
Use Sonar Branch Plugin ?

**Comment by dungdm93** [comment]
Yes. ` sonar-branch-plugin: 7.1`

**Comment by gabrie-allaigre** [comment]
It's a bug in version 3.0.0 gitlab plugin, 
use 3.0.1, only download

**Comment by dungdm93** [comment]
Thank you @gabrie-allaigre.
It helped me. But why don't you publish this patch to marketplace?

**Comment by TafThorne** [comment]
I think I am seeing this.  As a workaround I have disabled my GitLab integration by removing the changset ID, project ID and so on from my GitLab job.  This causes the integration to be skipped and the basic SonarQube analysis to complete.  The downside is that none of the code smells are mentioned back in GitLab and the GitLab CI will allow smelly code to pass and get merged back in.
